### PR TITLE
Clarify `gen_statem:start_link/3,4` semantics

### PR DESCRIPTION
**Pull request by RaimoNiskanen**
Solves #7524

### CODE REVIEW

File: lib/stdlib/doc/src/gen_statem.xml [hunk 1]
```diff
@@ -2472,31 +2472,41 @@ handle_event(_, _, State, Data) ->
...
+	  <seetype marker="#init_result"><c>{error,Reason}</c></seetype>
+	  the process is terminated and this function returns
+	  <seetype marker="#start_ret"><c>{error,Reason}</c></seetype>.
+	  If <c>Module:init/1</c> returns
+	  <seetype marker="#init_result"><c>ignore</c></seetype>
+	  the process is terminated and this function returns
+	  <seetype marker="#start_ret"><c>ignore</c></seetype>.
+        </p>
+        <p>
+          The process exits with <c>Reason</c> unless
```
Comment (attah) [inline lib/stdlib/doc/src/gen_statem.xml:2494]: Could do with re-asserting that what "the" process is here. At least my reference had expired once i read this far.

File: lib/stdlib/doc/src/gen_statem.xml [hunk 2]
```diff
@@ -2472,31 +2472,41 @@ handle_event(_, _, State, Data) ->
...
+          <seemfa marker="#Module:init/1"><c>Module:init/1</c></seemfa>
+          returns <c>ignore</c> or <c>{error,_}</c>,
+          then the exit reason is <c>normal</c>.
+        </p>
+        <p>
+          If <c>start_link/3,4</c> returns <c>ignore</c> or <c>{error,_}</c>,
```
Comment (attah) [inline lib/stdlib/doc/src/gen_statem.xml:2500]: Would it be useful to point to the change notice from here or have a "since OTP-26"?

**Review by attah — COMMENTED** [review]
Definite improvement. Two minor suggestions though.

File: lib/stdlib/doc/src/gen_statem.xml (hunk 2 above)
Comment (RaimoNiskanen) [reply lib/stdlib/doc/src/gen_statem.xml:2500]: Absolutely.  I will even add a warning note.

**Review by kikofernandez — APPROVED** [review]

**Pull request merged**